zio-kafka/src/main/scala/zio/kafka/consumer/SubscriptionStreamControl.scala

```diff
@@ -0,0 +1,32 @@
+package zio.kafka.consumer
+import zio.UIO
+
+/**
+ * Allows graceful shutdown of a stream, where no more records are being fetched but the in-flight records can continue
+ * to be processed and their offsets committed.
+ *
+ * As long as this object is in scope, the Kafka consumer remains subscribed.
+ *
+ * @tparam StreamType
+ *   Type of the stream returned from [[stream]]
+ */
+trait SubscriptionStreamControl[StreamType] {
```

Comment:
`StreamType`, somehow I find it ugly to have the word `Type` in there. Scala style guide recommends we use a single letter. E.g. `S`.

Do we need to / can we restrict the type?

```suggestion
trait SubscriptionStreamControl[S <: X] {
```

for some type `X`